addons/github/tests/test_models.py

```diff
@@ -98,6 +98,25 @@ def test_to_json_user_is_not_owner(self, mock_org, mock_repos):
         assert_equal(result.get('repo_names', None), None)
 
 
+    @mock.patch('addons.github.api.GitHubClient.repos')
+    @mock.patch('addons.github.api.GitHubClient.my_org_repos')
+    def test_get_folders(self, mock_org, mock_repos):
+        mock_repos.return_value = [Repository.from_json({'name': 'test',
+                                                         'id': '12345',
+                                                         'owner':
+                                                             {'login': 'test name'}
+                                                         })
+                                   ]
+        mock_org.return_value = {}
+        result = self.node_settings.get_folders()
+
+        assert_equal(len(result), 1)
+        assert_equal(result[0]['id'], '12345')
+        assert_equal(result[0]['name'], 'test')
+        assert_equal(result[0]['path'], 'test name/test')
+        assert_equal(result[0]['kind'], 'repo')
+
+
```

Comment:
Could you add a test for when the user does not have permission to view the folders?

Reply:
Tests for that are done in  api_tests/nodes/views/test_node_addons.py as subclasses of a base test class.